crates/turbopath/src/absolute_system_path.rs

```diff
@@ -0,0 +1,176 @@
+#[cfg(not(windows))]
+use std::os::unix::fs::symlink as symlink_file;
+#[cfg(not(windows))]
+use std::os::unix::fs::symlink as symlink_dir;
+#[cfg(windows)]
+use std::os::windows::fs::{symlink_dir, symlink_file};
+use std::{
+    borrow::Cow,
+    fmt, fs,
+    fs::Metadata,
+    io,
+    path::{Path, PathBuf},
+};
+
+use path_slash::PathExt;
+
+use crate::{
+    AbsoluteSystemPathBuf, AnchoredSystemPathBuf, IntoSystem, PathError, PathValidationError,
+    RelativeSystemPathBuf, RelativeUnixPath,
+};
+
+pub struct AbsoluteSystemPath(Path);
+
+impl ToOwned for AbsoluteSystemPath {
+    type Owned = AbsoluteSystemPathBuf;
+
+    fn to_owned(&self) -> Self::Owned {
+        AbsoluteSystemPathBuf(self.0.to_owned())
+    }
+}
+
+impl AsRef<AbsoluteSystemPath> for AbsoluteSystemPath {
+    fn as_ref(&self) -> &AbsoluteSystemPath {
+        self
+    }
+}
+
+impl fmt::Display for AbsoluteSystemPath {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        self.0.display().fmt(f)
+    }
+}
+
+impl AsRef<Path> for AbsoluteSystemPath {
+    fn as_ref(&self) -> &Path {
+        &self.0
+    }
+}
+
+impl AbsoluteSystemPath {
+    pub fn new<P: AsRef<Path> + ?Sized>(value: &P) -> Result<Cow<Self>, PathError> {
+        let path = value.as_ref();
+        if path.is_relative() {
+            return Err(PathValidationError::NotAbsolute(path.to_owned()).into());
+        }
+
+        let system_path = path.to_slash().ok_or_else(|| {
```

Comment:
I don't think this is right. `to_slash` converts to unix-style, not system.

Separately, I'm not sure we want `new` to do the conversion. I think having that be explicit would be helpful for readers following the code that uses this type.

Reply:
Do we want to have an unchecked new, or just allow creating an `AbsoluteSystemPath` from an `AbsoluteSystemPathBuf`?

Reply:
I suspect this is worthing having because most paths that we encounter will be system paths, so they won't require any allocation. Right now we do a copy every time we wrap something in `AbsoluteSystemPathBuf`

Reply:
I think a `pub(crate) fn unchecked_new(...)` would be appropriate for places inside `turbopath` where we already know we have a system path.

Reply:
And if you find yourself needing to convert to a system path from _outside_ of `turbopath`, I think an explicit call to a conversion function is the way to do it, even if it returns a `Cow` of some sort.

Reply:
Isn't this an explicit call to a conversion function? I'm not clear on how that's different than what we have here

Reply:
Discussed in person, but to document: we should have a constructor where we are validating but not converting for instances where we expect to already have an absolute system path, and a constructor that explicitly does the conversion if we have places in the code where we need to do a conversion.